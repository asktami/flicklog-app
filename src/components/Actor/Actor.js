import React from 'react';
import MISSING_IMG from '../../img/missing_image.png';

const Actor = ({ id, name, character, profile_path }) => {
	let pic = profile_path
		? 'https://image.tmdb.org/t/p/w138_and_h175_face' + profile_path
		: MISSING_IMG;

	return (
		<div className="actor">
			<a
				href={`https://www.themoviedb.org/person/${id}`}
				target="_blank"
				rel="noopener noreferrer"
			>
				<img src={pic} alt={name} />
			</a>
			<br />
			<span className="actor-name">{name}</span>
			<br />
			<span className="actor-character">{character}</span>
		</div>
	);
};

export default Actor;
