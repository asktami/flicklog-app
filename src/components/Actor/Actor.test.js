import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Actor from './Actor';

describe(`Actor component`, () => {
	it('renders a Actor by default', () => {
		const wrapper = shallow(<Actor />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
