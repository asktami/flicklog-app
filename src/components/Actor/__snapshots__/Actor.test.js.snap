// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actor component renders a Actor by default 1`] = `
<div
  className="actor"
>
  <a
    href="https://www.themoviedb.org/person/undefined"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      src="missing_image.png"
    />
  </a>
  <br />
  <strong />
  <br />
  <em />
</div>
`;
